javaagent-api/src/main/jflex/SqlSanitizer.flex

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.api.db.sanitizer;
+
+import io.opentelemetry.javaagent.instrumentation.api.db.SqlStatementInfo;
+
+%%
+
+%public
+%final
+%class SqlSanitizer
+%apiprivate
+%int
+%buffer 2048
+
+%unicode
+%ignorecase
+
+COMMA             = ","
+OPEN_PAREN        = "("
+CLOSE_PAREN       = ")"
+OPEN_COMMENT      = "/*"
+CLOSE_COMMENT     = "*/"
+IDENTIFIER        = ([:letter:] | "_") ([:letter:] | [0-9] | [_.])*
+BASIC_NUM         = [.+-]* [0-9] ([0-9] | [eE.+-])*
+HEX_NUM           = "0x" ([a-f] | [A-F] | [0-9])+
+QUOTED_STR        = "'" ("''" | [^'])* "'"
+DOUBLE_QUOTED_STR = "\"" ("\"\"" | [^\"])* "\""
+DOLLAR_QUOTED_STR = "$$" [^$]* "$$"
+WHITESPACE        = [ \t\r\n]+
+
+%{
+  public static SqlStatementInfo sanitize(String statement) {
```

Comment:
I was already planning to do something similar in the next PR, because I don't want to use `Config` inside generated code - but I guess I can just do it here.